system_tests/bigquery.py

```diff
@@ -90,7 +92,15 @@ def test_update_dataset(self):
         after = [grant for grant in dataset.access_grants
                  if grant.entity_id != 'projectWriters']
         dataset.access_grants = after
-        dataset.update()
+
+        # We should try and keep `delay` and tries as low as possible to
+        # reduce test running time. tries=3 and delay=30 worked consistenly
+        # when updating a dataset.
```

Comment:
Ah, so all this is doing is updating the permissions on the dataset AFAICT. Which I would think shouldn't be causing this rate limit error to pop up. It really doesn't make sense to me that it's an issue here.

Do you have any ideas?